edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/ServiceIdentityDictionary.cs

```diff
@@ -25,7 +25,7 @@ public ServiceIdentityDictionary(string actorDeviceId)
 
         public string GetActorDeviceId() => this.actorDeviceId;
 
-        public Task<Option<string>> GetAuthChain(string id) => throw new NotImplementedException("Nested Edge not enabled");
+        public Task<Option<string>> GetAuthChain(string id) => Task.FromResult(Option.None<string>());
```

Comment:
Per `\edge-hub\core\src\Microsoft.Azure.Devices.Edge.Hub.CloudProxy\ServiceProxy.cs` line 115 and line 47, why do they think the nested edge is enabled? That shouldn't be the case here for master branch E2E. 
@richma-ms Thoughts on this?

Reply:
In the case I'm fixing, they don't think nested edge is enabled - GetAuthChain is called by ScopeIdentitiesHandler when sending over scopes to broker, as a part of the normal, non-nested case. So GetAuthChain should not throw